src/libsyntax_pos/lib.rs

```diff
@@ -532,26 +604,55 @@ impl FileMap {
         lines.push(pos);
     }
 
+    /// add externally loaded source.
```

Comment:
Oh and this comment should probably mention how it shouldn't be called except from `CodeMap::load_source_for_filemap`.